corehq/apps/locations/models.py

```diff
@@ -353,7 +353,7 @@ class SQLLocation(MPTTModel):
     metadata = jsonfield.JSONField(default=dict, blank=True)
     created_at = models.DateTimeField(auto_now_add=True)
     last_modified = models.DateTimeField(auto_now=True)
-    is_archived = models.BooleanField(default=False)
+    is_archived = models.BooleanField(default=False, db_index=True)
```

Comment:
```
SELECT
    "commtrack_stockstate"."id",
    "commtrack_stockstate"."section_id",
    "commtrack_stockstate"."case_id",
    "commtrack_stockstate"."product_id",
    "commtrack_stockstate"."stock_on_hand",
    "commtrack_stockstate"."daily_consumption",
    "commtrack_stockstate"."last_modified_date",
    "commtrack_stockstate"."last_modified_form_id",
    "commtrack_stockstate"."sql_product_id",
    "commtrack_stockstate"."sql_location_id"
FROM
    "commtrack_stockstate"
    INNER JOIN "products_sqlproduct" ON ("commtrack_stockstate"."sql_product_id" = "products_sqlproduct"."id")
        LEFT OUTER JOIN "locations_sqllocation" ON ("commtrack_stockstate"."sql_location_id" = "locations_sqllocation"."id")
    WHERE (NOT ("products_sqlproduct"."is_archived" = TRUE)
        AND NOT ("locations_sqllocation"."is_archived" = TRUE
            AND "locations_sqllocation"."is_archived" IS NOT NULL)
        AND "commtrack_stockstate"."case_id" IN (...));
```

I'm not sure if this will help, but it looked like it could based on this query. i'm sure there's a better way to write this but, I don't have any idea where this comes from at the moment. it's just hte output from slowlogs in pgbadger.